From cb030ec1285d2f524822f39c823d3bb1581dedd0 Mon Sep 17 00:00:00 2001
From: Kyle Taylor <kyletaylored@gmail.com>
Date: Fri, 26 Sep 2014 02:41:24 -0500
Subject: [PATCH] Add support for population of alt/title attributes.

---
 plugins/mediaBrowser/plugin.js | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/plugins/mediaBrowser/plugin.js b/plugins/mediaBrowser/plugin.js
index aed6414..ffe2cfb 100644
--- a/plugins/mediaBrowser/plugin.js
+++ b/plugins/mediaBrowser/plugin.js
@@ -62,8 +62,28 @@
         // use that. For now, just remove the server name from the absolute
         // url to get the relative. FIX THIS!!!
         var fileUrl = mediaFiles[0].url.replace(location.origin, '');
+        var fileAlt = (mediaFiles[0].alt) ? mediaFiles[0].alt : "" ;
+        var fileTitle = (mediaFiles[0].title) ? mediaFiles[0].title : "";
+
         var parts = evt.sender.filebrowser.target.split(':');
         dialog.setValueOf(parts[0], parts[1], fileUrl);
+
+        // Add title/alt attr using default CKEditor image values.
+        var altEl = dialog.getContentElement("info", "txtAlt");
+        if (altEl) {
+          altEl.setValue(fileAlt);
+        }
+        // Try CKEditor defined title value first.
+        var titleEl = dialog.getContentElement("info", "txtGenTitle");
+        if (titleEl) {
+          titleEl.setValue(fileTitle);
+        } else {
+          // If it doesn't work, try a more generic ID.
+          titleEl = dialog.getContentElement("info", "txtTitle");
+          if (titleEl) {
+            titleEl.setValue(fileTitle);
+          }
+        }
       }
     });
     $(mediaIframe).parent().css({'z-index':'10002'});
-- 
1.9.3 (Apple Git-50)

